### ISSUE TITLE
On-board foundation tests to CI and nightly builds through VSO.

### ISSUE THREAD

**Issue by rajsesh-msft** [Microsoft/WinObjC#916]

**Comment by msft-Jeyaram** [comment]
The build definition is already setup to support the pull request.
Just need to add it to the xml file and test it

**Comment by msft-Jeyaram** [comment]
Blocked by OSX tests that are currently failing.

**Comment by aballway** [comment]
Failing tests on the reference platform were fixed as of abe4cc244467efecfc618a03348b64af27ed2cfe so this is no longer blocked 🛩

**Comment by msft-Jeyaram** [comment]
Yay, I'll get that up and running today :)

**Comment by msft-Jeyaram** [comment]
This is a duplicate of #2645.
We have support to build and run the OSX unit tests as part of the release pipelines.
The OSX tests are running and passing (YAY)

This will be part of the daily builds. Going forward #2645 address any enhanced work.
